### ISSUE TITLE
Colored Signs & Another way of editing them

### ISSUE THREAD

**Issue by D3nnis3n** [PrinceOfAmber/Cyclic#789]
Hello,
got some feedback from players today:

1) They would like to be able to edit signs without left clicking, as in creative this will remove the sign. Shift-Rightclick or Rightclick with a sign in hand would be preferred. Maybe something to think about?

2) There is no mod that i found yet that allows colorizing / formatting contents on signs. We all know it was possible with bukkit plugins but there is nothing for 1.12 forge yet. As i see that you do add some small tweaks - what about this one? In Bukkit they used & instead of § and then colored the following text.

**Comment by PrinceOfAmber** [comment]
for #1 yeah maybe i will think about changing to left click, right doesnt make sense since right click is attack/break.

for #2, yeah it does seem like all those are in game, but the only work in certain places.  Like book and quill they work fine but not sign.  ill see how possible it is https://minecraft.gamepedia.com/Formatting_codes#Sample_Text

**Comment by D3nnis3n** [comment]
You mean to change to rightclick? Left click is the actual behaviour atm.

Sam <notifications@github.com> schrieb am Mo., 28. Mai 2018, 07:39:

**Comment by D3nnis3n** [comment]
Nice.
Well, in Bukkit you just used $ instead of § for the color codes in text. That'd be enough for me.

You also could add Color-Buttons or something else - but i'm not sure if that's worth it.

**Comment by PrinceOfAmber** [comment]
Ok no keybinds.

Looks like vanilla sign rendering also supports the following options on top of colour

`    private Boolean bold;
    private Boolean italic;
    private Boolean underlined;
    private Boolean strikethrough;
    private Boolean obfuscated; // not sure what this means yet
`

**Comment by D3nnis3n** [comment]
Obufscated makes the text to runes like in the enchantment table, while those runes are changing every second. It’s useless, but funny.

Gesendet von Mail für Windows 10

Von: Sam
Gesendet: Montag, 28. Mai 2018 22:38
An: PrinceOfAmber/Cyclic
Cc: D3nnis3n; Author
Betreff: Re: [PrinceOfAmber/Cyclic] Colored Signs & Another way of editingthem (#789)

Ok no keybinds.
Looks like vanilla sign rendering also supports the following options on top of colour
private Boolean bold; private Boolean italic; private Boolean underlined; private Boolean strikethrough; private Boolean obfuscated; // not sure what this means yet

**Comment by PrinceOfAmber** [comment]
a rainbow crayon

![sign editor](https://user-images.githubusercontent.com/621529/40704354-aef4ab6c-639c-11e8-8565-df689e4933d5.gif)

**Comment by PrinceOfAmber** [comment]
Yeah i wanted to make sure to save it in the vanilla data format, otherwise it seemed like it might involve more hacking into sign rendderer (unless im wrong)

For example it has data like this

https://minecraft.gamepedia.com/Sign


`{Text2:"{\"text\":\"line2\",\"color\":\"blue\"}"}`

so Text2 means this is line 2, and then it has one text tag for the line and one color tag
